components/Button/SwitchNetwork.tsx

```diff
@@ -0,0 +1,37 @@
+import { Button, ButtonProps } from '@chakra-ui/react'
+import { PropsWithChildren, useCallback } from 'react'
+import { useChainId, useSwitchNetwork } from 'wagmi'
+
+const ButtonWithNetworkSwitch = ({
+  children,
+  chainId,
+  ...props
+}: PropsWithChildren<
+  ButtonProps & {
+    chainId: number
+  }
+>): JSX.Element => {
+  const currentChainId = useChainId()
+  const { switchNetwork } = useSwitchNetwork()
+
+  const handleSwitchNetwork = useCallback(() => {
+    if (!switchNetwork) return
+    switchNetwork()
+  }, [switchNetwork])
+
+  if (currentChainId !== chainId)
+    return (
+      <Button
+        {...props}
+        onClick={handleSwitchNetwork}
+        type="button"
+        leftIcon={undefined}
+        rightIcon={undefined}
+      >
+        Switch Network
```

Comment:
Could we show the network the user will be switched to?
Better UX and if there is an error, the user will be able to switch manually to the right network, now that it can read the network name.
```suggestion
        Switch to ${networks[chainId].name}
```

Reply:
This starts to be really big for some action button

Reply:
done in https://github.com/liteflow-labs/starter-kit/pull/194